tests/performance/index.js

```diff
@@ -11,19 +18,24 @@ function runTestSuites(PouchDB) {
     (adapters.length > 0 ? (', using adapter(s): ' + adapters.join(', ')) : '') +
     '\n\n');
 
+  const suites = (commonUtils.params().suites && commonUtils.params().suites.split(',')) || ALL_SUITES;
+  if (suites.some(s => !ALL_SUITES.includes(s))) {
+    throw new Error(`Unrecongnised suite(s): '${suites}'`);
+  }
+
   var theAdapterUsed;
   var count = 0;
   function checkDone(adapterUsed) {
     theAdapterUsed = theAdapterUsed || adapterUsed;
-    if (++count === 4) { // number of perf.xxxx.js tests
+    if (++count === suites.length) {
       reporter.complete(theAdapterUsed);
     }
   }
 
-  require('./perf.basics')(PouchDB, checkDone);
-  require('./perf.views')(PouchDB, checkDone);
-  require('./perf.find')(PouchDB, checkDone);
-  require('./perf.attachments')(PouchDB, checkDone);
+  if (suites.includes('basics')) { require('./perf.basics')(PouchDB, checkDone); }
+  if (suites.includes('views')) { require('./perf.views')(PouchDB, checkDone); }
+  if (suites.includes('find')) { require('./perf.find')(PouchDB, checkDone); }
+  if (suites.includes('attachments')) { require('./perf.attachments')(PouchDB, checkDone); }
```

Comment:
Consider iterating over ALL_SUITES? For example, change ALL_SUITES to an obj with name key and path value?

```js
const ALL_SUITES = {
  basics: './perf.basics',
  views: './perf.views',
  find: './perf.find',
  attachments: './perf.attachments',
};

for (const [name, filepath] of Object.entries(ALL_SUITES)) {
  if (suites.includes(name)) {
    require(filepath)(PouchDB, checkDone);
  }
}
```

Reply:
This looks much clearer, and was similar to my initial approach, but my memory is that dynamic `require(...)` calls were breaking bundling.  

I've tried recreating the issue at https://github.com/alxndrsn/pouchdb/compare/perf-tests-suite-filter...alxndrsn:pouchdb:perf-tests-suite-filter-indirect-requires?expand=1, but it seems to build ok: https://github.com/alxndrsn/pouchdb/actions/runs/4890727694

However, I'm nervous to make the change as I'm not confident that it will work in all situations :grimacing:

Reply:
Ok. Feel free to merge as is!